### PR TITLE
added passphrase to audit-board based on english words, URL slug redirection

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -468,7 +471,8 @@ def jurisdictions_set(election_id):
             audit_board_obj = AuditBoard(
                 id = audit_board["id"],
                 name = audit_board["name"],
-                jurisdiction_id = jurisdiction_obj.id
+                jurisdiction_id = jurisdiction_obj.id,
+                passphrase = xp.generate_xkcdpassword(WORDS, numwords=4, delimiter="-")
```
Comment (eventualbuddha) [inline app.py]: Just to clarify, we don't care that these will be different given the same seed, right?

**Comment by MorganLove** [comment]
#232 has the frontend corollary to this by including the passphrase link in the downloadable PDF.

File: models.py [hunk 2]
```diff
@@ -96,6 +96,7 @@ class AuditBoard(db.Model):
     member_1_affiliation = db.Column(db.String(200), nullable=True)
     member_2 = db.Column(db.String(200), nullable=True)
     member_2_affiliation = db.Column(db.String(200), nullable=True)
+    passphrase = db.Column(db.String(1000), nullable=True)
```
Comment (eventualbuddha) [inline models.py]: Should there be a unique index on this?

Comment (benadida) [reply models.py]: yeah, there probably should, and should also check for error conditions, will add those.

File: app.py (hunk 1 above)
Comment (benadida) [reply app.py]: correct, the consistency we care about is for the audit data itself, not for authentication.

**Comment by benadida** [comment]
added the unique constraint. I don't think we need any additional error checking for now. On the offchance we hit a duplication, the jurisdiction set call will fail and roll back the transaction. If we hit it for real, we can make the error a bit nicer.

**Comment by benadida** [comment]
Ah, we've gotten ourselves in a bit of a merge mess here. @MorganLove let's sync up and see if we can disentangle this. I think the printed credential URLs are the old URLs, not the passphrase ones, and I'm having trouble disentangling the commits.

**Comment by MorganLove** [comment]
> Ah, we've gotten ourselves in a bit of a merge mess here. @MorganLove let's sync up and see if we can disentangle this. I think the printed credential URLs are the old URLs, not the passphrase ones, and I'm having trouble disentangling the commits.

@benadida I resolved the merge conflict in the end to end test file, and double checked to make sure the printed url is the one with the passphrase, [and it is](https://github.com/votingworks/arlo/blob/0ee75dd772c0be87e843da1265f311384c594505/arlo-client/src/components/AuditForms/CalculateRiskMeasurement.tsx#L236). We should be good to go then. 🙂

**Comment by eventualbuddha** [comment]
As for rebasing & merging issues, I rebased this branch onto master: `generate-audit-board-slugs-rebased`. It seems to be identical to this branch in that `git diff` is empty. Feel free to force-push that branch to this one if you want to use it @benadida or @MorganLove.

**Comment by MorganLove** [comment]
> As for rebasing & merging issues, I rebased this branch onto master: `generate-audit-board-slugs-rebased`. It seems to be identical to this branch in that `git diff` is empty. Feel free to force-push that branch to this one if you want to use it @benadida or @MorganLove.

@eventualbuddha It was the commit history that was mangled up, so I went back in and did an interactive rebase to clean it up and force pushed that. Thanks though!

@benadida Good catch on the git history. Should we set that up as a standard part of the flow, so that we always remember to do an interactive rebase to clean up the commits before merging?

**Pull request merged**